compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal

```diff
@@ -590,6 +623,10 @@ function loadType(jvm:MethodVisitor mv, bir:BType? bType) {
     } else if (bType is bir:BTypeDesc) {
         typeFieldName = "typeTypedesc";
     }  else if (bType is bir:BServiceType) {
+        if (getTypeFieldName(bType.oType.name.value) != "$type$service") {
```

Comment:
even here, i think it's better to unify(like write default service type as an object type, so you don't need to differentiate here, anyway do that refactoring later with a different PR)